### PR TITLE
Controller + Template Composite Primary Key Support

### PR DESCRIPTION
**Pull request by ionas**
Moved and improved upon https://github.com/cakephp/bake/pull/180

### CODE REVIEW

**Comment by ionas** [comment]
On this issue please help me with the test cases because it is weirdo and not regular test cases. What to do exactly and how?

File: src/Template/Bake/Element/Controller/edit.ctp [hunk 1]
```diff
@@ -21,13 +23,29 @@ $compact = ["'" . $singularName . "'"];
...
+}
+%>
+    public function edit(<%= join($actionArguments, ', ') %>)
     {
-        $<%= $singularName %> = $this-><%= $currentModelName %>->get($id, [
+        <%- 
+        $primaryKeyArguments = [];
+        foreach ($primaryKeys as $primaryKeyComponent) {
+            $primaryKeyArguments[] = '$' . Inflector::variable($primaryKeyComponent);
+        }
```
Comment (markstory) [inline src/Template/Bake/Element/Controller/edit.ctp]: Isn't this the same as `actionArguments`?

File: src/Template/Bake/Template/view.ctp [hunk 2]
```diff
@@ -49,23 +49,33 @@ $groupedFields = collection($fields)
     ->toArray();
 
 $groupedFields += ['number' => [], 'string' => [], 'boolean' => [], 'date' => [], 'text' => []];
-$pk = "\$$singularVar->{$primaryKey[0]}";
+$primaryKeyArguments = [];
+foreach ($primaryKey as $primaryKeyComponent) {
+    $primaryKeyArguments[] = '$' . $singularVar . '->' . $primaryKeyComponent;
+}
+$primaryKeyArgumentList = join($primaryKeyArguments, ', ');
```
Comment (markstory) [inline src/Template/Bake/Template/view.ctp]: This snippet is in a few templates. Is it worth adding a helper method for this?

File: src/Template/Bake/Element/Controller/edit.ctp (hunk 1 above)
Comment (ionas) [inline src/Template/Bake/Element/Controller/edit.ctp]: No the actionArguments have ` . ' = null';`, else they are the same.

File: src/Template/Bake/Template/view.ctp (hunk 2 above)
Comment (ionas) [inline src/Template/Bake/Template/view.ctp]: It exists twice. Here and in form.ctp. I'd not know where to add the helper method.

**Comment by ionas** [comment]
I am totally "out" of this PR now (took some time until it got reviewed and I got a lot of deadlines and I am still pushing them).

My test environment for this is gone - so I'd need help with testing and creating test cases. In case any of you are interested in picking it up OR want to help me out, please reach out to me. I'll close this for now as I can't see when I'll get to finishing it solely on my own.

**Pull request closed without merging**